src/receive_imf.rs

```diff
@@ -2841,6 +2841,16 @@ async fn apply_group_changes(
         if !is_from_in_chat {
             better_msg = Some(String::new());
         } else if let Some(key) = mime_parser.gossiped_keys.get(added_addr) {
+            if !chat_contacts.contains(&from_id) && !chat_contacts.contains(&ContactId::SELF) {
```

Comment:
```suggestion
            if !chat_contacts.contains(&from_id) {
```
The second part is always true, see the expression for `is_from_in_chat` above. Anyway better to remove it because we already decided that `from_id` is in the chat, but it's not in `chat_contacts`, so this should be fixed up (this is how the code can be interpreted)